### ISSUE TITLE
Giving kudos stops working

### ISSUE THREAD

**Issue by monteriding** [mjaschen/strava-helper-firefox#35]
The give kudos to all (and giving kudos one by one) appears to stop working if giving out a large number of kudos (several hundred to several clubs). This issue prevents giving kudos one by one or kudos to all once it occurs, and it requires closing out of the browser and waiting for a period of time before it works again. Once the issue occurs, giving kudos in a different browser stops working also.

**Comment by mjaschen** [comment]
Hi!

> once it occurs, and it requires closing out of the browser and waiting for a period of time 

> Once the issue occurs, giving kudos in a different browser stops working also.

I'm pretty sure that you reached an API rate limiting for your Strava account so that all your “Give Kudos” requests are blocked by Strava. You can check the Console in the Developer Tools (you can open it by pressing the F12 key) to confirm my guess.

Marcus

**Comment by monteriding** [comment]
Marcus,

Thanks for the response. You're correct, it does look like that is what's
happening -

[image: Inline image 1]

This is fairly recent that Strava implemented this (it didn't do this two
weeks ago). Any way to work around this, or introduce a (perhaps
configurable) delay between posts that shouldn't trigger the Strava API
rate limit?

thx :)
monte

**Comment by mjaschen** [comment]
It seems that Strava has increased the API request limits in the meanwhile, as I'm unable to observe this behaviour anymore. Can anybody confirm this?

**Comment by monteriding** [comment]
Yes, I'm still experiencing it, but not as frequently. It seems to be
allowing about double what it's been in the recent past.

**Comment by o2dazone** [comment]
hey @mjaschen - I don't use your addon, but I develop a Chrome extension that does something similar to your auto-Kudos feature. I've had better success against throttling by querying the DOM for kudos, and then on an interval, trigger a click on each element. I've also considered toying with randomizing the frequency by an additional 200-1000ms so as not to get flagged by any automation thwarts they attempt to do in the future.

Anyway, hope it helps! Nice code btw!

**Comment by tardoin** [comment]
Edit, I was in the wrong site. Please disregard. Not a problem with Strava Helper.

~~Did they just lower this a lot? I'm running into some kind of limit. I give a decent amount of kudos -- by hand, probably tried to do like 30-40ish or something before it started to not work, after being away for a couple days. If that's what's going on, it's a terribly implemented limit! It doesn't really even pop anything up to say what's going on. Come on, Strava.~~

~~In the console, I don't really see an error, but I do see an entry that looks like this when I try (and it fails) to kudos someone:
VM1687:3698 POST https://www.strava.com/feed/athlete-feed-entry/[redacted]/kudo 429~~

~~This is kind of BS, don't just silently change it and have it silently fail and waste my and everyone's time!~~

~~Also I did open my phone and finish the job, so take that.~~

~~Maybe it's just Chrome or something, but I kind of doubt it because it was doing it last night and started again today. Working fine for awhile then just stops turning the thumbs-up icon red even though it kind of looks like it wants to work. I'm reading my friends' activities, not watching to verify something that should work actually worked, so I get kind of far before I notice it's not working.~~

**Comment by mjaschen** [comment]
Once in a while Strava tightens the rate-limiting, usually it works again after a few days. I also get HTTP status 429 responses at the moment.

I'll watch this and try limiting the request rate on the client side as a workaround if the behavior doesn't change in the next days.

**Comment by tardoin** [comment]
Whoah, for some reason I thought I was in Strava support. Duh. Sorry! Not your fault at all.

**Comment by jus2deep** [comment]
Would it be possible to create a way to automate kudos in the Strava android app? Especially since it seems Strava only limits kudos from browsers.
